src/View/AutoSegmentation/SegmentSelectorWidget.py

```diff
@@ -46,22 +46,25 @@ def __init__(self, parent, segmentation_list: list[str], data_location="data/csv
         # Reference to the list owned by another class most likely the controller class
         self._selected_list: list[str] = segmentation_list
         # References to look up items in tree without searching entire tree
-        self._tree_choices_ref: dict[str, QTreeWidgetItem] = {}
+        self._tree_choices_ref: dict[str, list[QTreeWidgetItem]] = {}
 
         # Creating Tree using PySide6
         # Nesting methods here to Indicate that each one is intended to feed into each other
         self._selection_tree = self._resize_columns(
             self._enter_tree_data(csv_location=pathlib.Path(data_location)/"segmentation_lists.csv", tree=self._create_selection_tree(
                 self._body_section_clicked)))
 
+        # Callback Methods
+        self.update_save_start = update_callback
```

Comment:
this seems important, but it's very unclear what will get saved, and when it actually happens.